gensim/models/ldaseqmodel.py

```diff
@@ -0,0 +1,1121 @@
+        mean[t] = w * fwd_mean[t] + (1 - w) * mean[t + 1]
+        if mean[t] is None:
+            # error message
+            pass
+ 
+    return
+
+def compute_expected_log_prob(sslm):
+
+    W = sslm.num_terms
+    T = sslm.num_sequences
+    for t in range(0, T):
+        for w in range(0, W):
+            sslm.e_log_prob[w][t] = sslm.mean[w][t + 1] - numpy.log(sslm.zeta[t])
+    return
+
+
+def sslm_counts_init(sslm, obs_variance, chain_variance, sstats):
+
+    W = sslm.num_terms
+    T = sslm.num_sequences
+
+    log_norm_counts = numpy.copy(sstats)
+    log_norm_counts = log_norm_counts / sum(log_norm_counts)
+
+    log_norm_counts = log_norm_counts + 1.0 / W
+    log_norm_counts = log_norm_counts / sum(log_norm_counts)
+    log_norm_counts = numpy.log(log_norm_counts)
+    
+
+    # setting variational observations to transformed counts
+    for t in range(0, T):
+        sslm.obs[:,t] = log_norm_counts
+
+    # set variational parameters
+    sslm.obs_variance = obs_variance
+    sslm.chain_variance = chain_variance
+
+    # compute post variance
+    for w in range(0, W):
+       compute_post_variance(w, sslm, sslm.chain_variance)
+
+    for w in range(0, W):
+       compute_post_mean(w, sslm, sslm.chain_variance)
+
+    update_zeta(sslm)
+    compute_expected_log_prob(sslm)
+
+def init_ldaseq_ss(ldaseq, topic_chain_variance, topic_obs_variance, alpha, init_suffstats):
+
+    ldaseq.alphas = alpha
+    for k in range(0, ldaseq.num_topics):
+
+        sstats = init_suffstats[:,k]
+        sslm_counts_init(ldaseq.topic_chains[k], topic_obs_variance, topic_chain_variance, sstats)
+
+        # initialize the below matrices only if running DIM
+        # ldaseq.topic_chains[k].w_phi_l = numpy.zeros((ldaseq.num_terms, ldaseq.num_sequences))
+        # ldaseq.topic_chains[k].w_phi_sum = numpy.zeros((ldaseq.num_terms, ldaseq.num_sequences))
+        # ldaseq.topic_chains[k].w_phi_sq = numpy.zeros((ldaseq.num_terms, ldaseq.num_sequences))
+
+def fit_lda_seq(ldaseq, seq_corpus):
+
+    K = ldaseq.num_topics
+    W = ldaseq.num_terms
+    data_len = seq_corpus.num_sequences
+    num_docs = seq_corpus.num_docs
+    
+    LDA_INFERENCE_MAX_ITER = 25
+
+    bound = 0
+    heldout_bound = 0
+    ldasqe_em_threshold = 1e-4
+    convergence = ldasqe_em_threshold + 1
+
+    iter_ = 0
+    final_iters_flag = 0 
+    last_iter = 0
+
+    # this is a flag/input do something about it
+    lda_seq_min_iter = 6
+    lda_seq_max_iter = 20
+    
+    while iter_ < lda_seq_min_iter or ((final_iters_flag is 0 or convergence > ldasqe_em_threshold) and iter_ <= lda_seq_max_iter):
+        if not (iter_ < lda_seq_min_iter or ((final_iters_flag is 0 or convergence > ldasqe_em_threshold) and iter_ <= lda_seq_max_iter)):
+            last_iter = 1
+
+        # log
+        print (" EM iter " , iter_)
+        print ("E Step")
+
+        # writing to file
+        em_log.write(str(bound) + "\t" + str(convergence))
```

Comment:
The C code writes these to disk. I'll just give the option of saving the model to disk and won't need this.